crates/iota-genesis-builder/src/stardust/process_outputs.rs

```diff
@@ -374,3 +643,81 @@ fn get_address_if_vesting_output(
         }
     })
 }
+
+// SwapSplit operation. Take a basic output and swap its address unlock
+// condition with a destination address. If  the basic output amount is greater
+// than the target, then the basic output is split in 2. In this case, the
+// second output keeps the original address unlock condition but its amount will
+// be reduced to the remainder of the operation.
+fn swap_split_operation<'a>(
+    destinations: impl Iterator<Item = (&'a mut IotaAddress, &'a mut u64)>,
+    basic_output: &BasicOutput,
+) -> (Option<Output>, Vec<Output>) {
+    let mut original_output_maybe = None;
+    let mut split_outputs_maybe = vec![];
+    let mut original_basic_output_remainder = basic_output.amount();
+
+    // if the addressUC's address is to swap, then it can have several
+    // destinations
+    for (destination, target) in destinations {
+        // break if the basic output was drained already
+        if original_basic_output_remainder == 0 {
+            break;
+        }
+        // we need to make sure that we split at most OUTPUT_INDEX_MAX - 1 times
+        if split_outputs_maybe.len() > OUTPUT_INDEX_MAX as usize - 1 {
+            panic!("Too many swap split operations to perform for a single output")
+        }
+        // if the target for this destination is less than the basic output remainder,
+        // then use it to split the basic output and swap address;
+        // otherwise split and swap using the original_basic_output_remainder, and then
+        // break the loop.
+        let swap_split_amount = cmp::min(*target, original_basic_output_remainder);
+        split_outputs_maybe.push(
+            BasicOutputBuilder::from(basic_output)
+                .with_amount(swap_split_amount)
+                .replace_unlock_condition(AddressUnlockCondition::new(Ed25519Address::new(
+                    destination.to_inner(),
+                )))
+                .finish()
+                .expect("should be able to create a basic output")
```

Comment:
Reword expects to descriptive? I.E. `.expect("failed to create basic output during split")`